### ISSUE TITLE
Error with pytim.ITIM

### ISSUE THREAD

**Issue by thangckt** [Marcello-Sega/pytim#345]
Dear Marcello, 

I try to run the following example code with pytim 0.8,4

    import MDAnalysis as mda
    import pytim
    from pytim.datafiles import WATER_GRO
    
    # load the configuration in MDAnalysis
    u = mda.Universe(WATER_GRO)
    
    # compute the interface using ITIM. Identify 4 layers.
    inter = pytim.GITIM(u,max_layers=4)
    #################

The code can run well, and produce result
But when I change to use:    

    pytim.ITIM(u,max_layers=4)

It throw the following Error:

    ###############
    ---------------------------------------------------------------------------
    NotImplementedError                       Traceback (most recent call last)
    <ipython-input-16-81c4521ead9c> in <module>()
          1 # initialize the interface, this has to be called only once
          2 # before iterating over the trajectory!
    ----> 3 interface = pytim.ITIM(mdaTraj)
          4 
          5 # write a pdb with the information on the surface layers
    
    C:\DevProgram\miniconda3\envs\py37\lib\site-packages\pytim\itim.py in __init__(self, universe, group, alpha, normal,     molecular, max_layers, radii_dict, cluster_cut, cluster_threshold_density, extra_cluster_groups, info, centered, warnings, mesh,     autoassign, **kargs)
        224         patchTrajectory(self.universe.trajectory, self)
        225 
    --> 226         self._assign_layers()
        227 
        228     def _create_mesh(self):
    
    C:\DevProgram\miniconda3\envs\py37\lib\site-packages\pytim\itim.py in _assign_layers(self)
        386 
        387         for p in proc:
    --> 388             p.start()
        389         for uplow in [up, low]:
        390             for index, group in enumerate(queue[uplow].get()):
    
    C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\process.py in start(self)
        110                'daemonic processes are not allowed to have children'
        111         _cleanup()
    --> 112         self._popen = self._Popen(self)
        113         self._sentinel = self._popen.sentinel
        114         # Avoid a refcycle if the target function holds an indirect
    
    C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\context.py in _Popen(process_obj)
        221     @staticmethod
        222     def _Popen(process_obj):
    --> 223         return _default_context.get_context().Process._Popen(process_obj)
        224 
        225 class DefaultContext(BaseContext):
    
    C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\context.py in _Popen(process_obj)
        320         def _Popen(process_obj):
        321             from .popen_spawn_win32 import Popen
    --> 322             return Popen(process_obj)
        323 
        324     class SpawnContext(BaseContext):
    
    C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\popen_spawn_win32.py in __init__(self, process_obj)
         87             try:
         88                 reduction.dump(prep_data, to_child)
    ---> 89                 reduction.dump(process_obj, to_child)
         90             finally:
         91                 set_spawning_popen(None)
        
    C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\reduction.py in dump(obj, file, protocol)
         58 def dump(obj, file, protocol=None):
         59     '''Replacement for pickle.dump() using ForkingPickler.'''
    ---> 60     ForkingPickler(file, protocol).dump(obj)
         61 
         62 #
    
    C:\DevProgram\miniconda3\envs\py37\lib\site-packages\MDAnalysis\core\universe.py in __getstate__(self)
        743 
        744     def __getstate__(self):
    --> 745         raise NotImplementedError
        746 
        747     def __setstate__(self, state):
        
    
    NotImplementedError: 
    ##############

Can you give me some hint to overcome this
Thank you so much

**Comment by Marcello-Sega** [comment]
Can you try if this works? 

     inter = pytim.ITIM(u,max_layers=4,multiproc=False)

**Comment by thangckt** [comment]
hi, 
This is error produce without multiproc

---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-5-aa2aee136c52> in <module>()
      7 
      8 # compute the interface using ITIM. Identify 4 layers.
----> 9 inter = pytim.ITIM(u,max_layers=4,multiproc=False)

C:\DevProgram\miniconda3\envs\py37\lib\site-packages\pytim\itim.py in __init__(self, universe, group, alpha, normal, molecular, max_layers, radii_dict, cluster_cut, cluster_threshold_density, extra_cluster_groups, info, centered, warnings, mesh, autoassign, **kargs)
    224         patchTrajectory(self.universe.trajectory, self)
    225 
--> 226         self._assign_layers()
    227 
    228     def _create_mesh(self):

C:\DevProgram\miniconda3\envs\py37\lib\site-packages\pytim\itim.py in _assign_layers(self)
    386 
    387         for p in proc:
--> 388             p.start()
    389         for uplow in [up, low]:
    390             for index, group in enumerate(queue[uplow].get()):

C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\process.py in start(self)
    110                'daemonic processes are not allowed to have children'
    111         _cleanup()
--> 112         self._popen = self._Popen(self)
    113         self._sentinel = self._popen.sentinel
    114         # Avoid a refcycle if the target function holds an indirect

C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\context.py in _Popen(process_obj)
    221     @staticmethod
    222     def _Popen(process_obj):
--> 223         return _default_context.get_context().Process._Popen(process_obj)
    224 
    225 class DefaultContext(BaseContext):

C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\context.py in _Popen(process_obj)
    320         def _Popen(process_obj):
    321             from .popen_spawn_win32 import Popen
--> 322             return Popen(process_obj)
    323 
    324     class SpawnContext(BaseContext):

C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\popen_spawn_win32.py in __init__(self, process_obj)
     87             try:
     88                 reduction.dump(prep_data, to_child)
---> 89                 reduction.dump(process_obj, to_child)
     90             finally:
     91                 set_spawning_popen(None)

C:\DevProgram\miniconda3\envs\py37\lib\multiprocessing\reduction.py in dump(obj, file, protocol)
     58 def dump(obj, file, protocol=None):
     59     '''Replacement for pickle.dump() using ForkingPickler.'''
---> 60     ForkingPickler(file, protocol).dump(obj)
     61 
     62 #

C:\DevProgram\miniconda3\envs\py37\lib\site-packages\MDAnalysis\core\universe.py in __getstate__(self)
    743 
    744     def __getstate__(self):
--> 745         raise NotImplementedError
    746 
    747     def __setstate__(self, state):

NotImplementedError:

**Comment by Marcello-Sega** [comment]
Hi,

It turns out, in some old commit the multiprocessing was set on by default and not controlled by the `multiproc` option anymore. I put it back in and the patch is currently being tested. Once it's merged, it would be great if you could try it out on your windows machine, let's see if this solves the issue.

**Comment by Marcello-Sega** [comment]
Merged. Could you please try with the latest version? (750c71d)

**Comment by thangckt** [comment]
Dear Marcello, 

I am installing pytim via: pip install 
So I don't know to install pytim from the source code. 
Can you please put a new release on PyPI?

And, maybe pytim.SASA also has the same multiproc-error as pytim.ITIM

**Comment by Marcello-Sega** [comment]
Hi,

no, I won't do that because if this does not fix the problem, I will have to keep bumping the version and keep uploading new binaries, and we cannot abuse PyPi as a debugging tool. 

To learn how to install the package from source, have a look at https://github.com/Marcello-Sega/pytim#installation  

M

**Comment by thangckt** [comment]
Dear Marcello,

Thank you for your update.
The pytim.ITIM work well now.
And as I mentioned, module pytim.SASA also has the same multiproc-error
Hope you consider this.

Thank you

**Comment by Marcello-Sega** [comment]
Hi,

sorry it took me so long. I just merged a new patch that should fix the SASA issue. 
Could you check it on your windows machine?

**Comment by thangckt** [comment]
Dear Marcello,

It runs now.
I have a problem with pytim.ITIM when I try to detect to free surface of the simple model as figure (free surface on 'z' and periodicity on 'x' and 'y')
![image](https://user-images.githubusercontent.com/46436648/104443931-70491000-55da-11eb-9b4c-b4a6a526bae3.png)
I use the following code:
##########
import MDAnalysis, pytim
mdaTraj = MDAnalysis.Universe('Model_Al_z001.cfg', format='lammpsdump')
inter = pytim.ITIM(mdaTraj, molecular=False)
inter.atoms
#######
and the result return: <AtomGroup with 0 atoms>

Can you give me some hints on what I get wrong?
I attracted the model file here: https://ufile.io/cmi0nxjs

Thank you so much

**Comment by Marcello-Sega** [comment]
let's move this to a new issue and close this one
